Core/Src/main.c

```diff
@@ -478,9 +478,13 @@ void task_MainTask(void *argument)
 {
   /* USER CODE BEGIN 5 */
   /* Infinite loop */
+
+  // Setting up variables...
+  DRIVE_STATE_t* curr_state = INV_LCKOUT;
+
   for(;;)
   {
-	  main_loop();
+	  curr_state = main_loop(curr_state);
```

Comment:
Another option here is to `extern` the `DRIVE_STATE_t` variable so that you can `vcu.c` without having to pass it in here